src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -332,8 +412,20 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
 
                 pBuffer += (bufferLength & MaskOfAllBitsInVector128) - SizeOfVector128;
 
-                currentMask = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
-                if (currentMask != 0)
+                if (Sse2.IsSupported)
+                {
+                    currentSseMaskOrAdvSimdIndex = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
```

Comment:
>currentSseMaskOrAdvSimdIndex [](start = 20, length = 28)

same here. It is little confusing to use same variable for SSE2 and AdvSimd that represents different things. Probably have dedicated variables for SSE2 and AdvSimd.